### PR TITLE
Find extension members should work in the background

### PR DESCRIPTION
**Pull request by nosami**
Fixes VSTS #636110 (#5118)

### CODE REVIEW

**Comment by nosami** [comment]
@mrward: I never noticed. I assume they are similarly affected. I'll check now.

**Comment by nosami** [comment]
I added a debug statement to check.
image

```
DEBUG [2018-06-21 10:29:58Z]: [UI - true] Finding derived symbols in /Users/jason/Projects/Roslyn.FSharp/Roslyn.FSharp/Roslyn.FSharp.fsproj : Symbol GetAttributeConstructorArguments
DEBUG [2018-06-21 10:29:58Z]: [UI - true] LanguageService: GetDerivedSymbolInProject: proj:/Users/jason/Projects/Roslyn.FSharp/Roslyn.FSharp/Roslyn.FSharp.fsproj, file:/Users/jason/Projects/Roslyn.FSharp/Roslyn.FSharp/ICompilation.fs, symbol:GetAttributeConstructorArguments
DEBUG [2018-06-21 10:29:58Z]: [UI - true] LanguageService: GetProjectCheckerOptions: Getting ProjectOptions from cache for:Roslyn.FSharp.fsproj}
```

I can see from the logging that quite a lot of work is being done on the UI thread. The other usages of StartFromContinuations aren't quite as obvious as they delegate to a background thread later on at some point. 

Really surprised that this hasn't been noticed before now. Also surprised that `Async.StartFromContinuations` doesn't start on a background thread given it's name.

File: main/external/fsharpbinding/MonoDevelop.FSharpBinding/RefactoringOperationsHandler.fs [hunk 1]
```diff
@@ -278,8 +278,9 @@ module Refactoring =
       with _ -> []
 
     let findReferences (editor:TextEditor, ctx:DocumentContext, symbolUse:FSharpSymbolUse, lastIdent) =
-        let monitor = IdeApp.Workbench.ProgressMonitors.GetSearchProgressMonitor (true, true)
-        let findAsync = async {
+        async {
+            use monitor = IdeApp.Workbench.ProgressMonitors.GetSearchProgressMonitor (true, true)
+            LoggingService.logDebug ("findReferences")
```
Comment (mrward) [inline main/external/fsharpbinding/MonoDevelop.FSharpBinding/RefactoringOperationsHandler.fs]: Debug logging left in here.

File: main/external/fsharpbinding/MonoDevelop.FSharpBinding/RefactoringOperationsHandler.fs [hunk 2]
```diff
@@ -278,8 +278,8 @@ module Refactoring =
       with _ -> []
 
     let findReferences (editor:TextEditor, ctx:DocumentContext, symbolUse:FSharpSymbolUse, lastIdent) =
-        let monitor = IdeApp.Workbench.ProgressMonitors.GetSearchProgressMonitor (true, true)
-        let findAsync = async {
+        async {
+            use monitor = IdeApp.Workbench.ProgressMonitors.GetSearchProgressMonitor (true, true)
```
Comment (Therzok) [inline main/external/fsharpbinding/MonoDevelop.FSharpBinding/RefactoringOperationsHandler.fs]: Shouldn't we still be creating/disposing the search monitor on the UI thread?

Comment (mrward) [reply main/external/fsharpbinding/MonoDevelop.FSharpBinding/RefactoringOperationsHandler.fs]: Can disposing be done on the background thread if the progress monitor is created with the correct synchronization context?

Comment (nosami) [reply main/external/fsharpbinding/MonoDevelop.FSharpBinding/RefactoringOperationsHandler.fs]: Ok. I'll change this

**Pull request merged**